common/network-common/src/main/java/org/apache/spark/network/server/ChunkFetchRequestHandler.java

```diff
@@ -124,7 +127,13 @@ private ChannelFuture respond(
       final Channel channel,
       final Encodable result) throws InterruptedException {
     final SocketAddress remoteAddress = channel.remoteAddress();
-    return channel.writeAndFlush(result).await().addListener((ChannelFutureListener) future -> {
+    ChannelFuture channelFuture = null;
+    if (chunkFetchAsyncModeEnabled) {
+      channelFuture = channel.writeAndFlush(result);
```

Comment:
When we remove `await`, we see more SASL requests timing out. I mentioned this in the comment here:
https://github.com/apache/spark/pull/22173#issuecomment-578347705

Wouldn't making the `asyncMode` default, make this problem worse?
In aysnc mode, how do you plan to tackle increased number of SASL failures?

Reply:
Thanks for the comment @otterc!
This PR aims to fix the performance regression brings by the sync mode fetching. As you mention, you find a problem with the stress test framework, I think the default mode should be the guarantee for common use cases.

For the SASL requests timing out, maybe we need more context for your internal stress test framework and analyze the root cause. IMO, there should be other configs in spark.shuffle/netty could help, not only depends on the async/sync mode here.